airflow/jobs/backfill_job.py

```diff
@@ -434,15 +433,15 @@ def _process_backfill_task_instances(
             # determined deadlocked while they are actually
             # waiting for their upstream to finish
             @provide_session
-            def _per_task_process(key, ti, session=None):
+            def _per_task_process(key, ti: TaskInstance, session=None):
                 ti.refresh_from_db(lock_for_update=True, session=session)
 
                 task = self.dag.get_task(ti.task_id, include_subdags=True)
                 ti.task = task
 
-                ignore_depends_on_past = self.ignore_first_depends_on_past and ti.execution_date == (
-                    start_date or ti.start_date
-                )
+                ignore_depends_on_past = self.ignore_first_depends_on_past and ti.get_dagrun(
+                    session
+                ).execution_date == (start_date or ti.start_date)
```

Comment:
```suggestion
                dagrun = self.ignore_first_depends_on_past and ti.get_dagrun(session=session)
                ignore_depends_on_past = dagrun.execution_date == (start_date or ti.start_date)
```

Gosh that Black formatting is awful.